src/components/document/Document.js

```diff
@@ -194,60 +300,125 @@ function Document({ docUrl, data, loading, fullPage, showToolbar = true, showMet
                 </Toolbar>
             )}
 
-            <TagsSection
-                loading={tagsLoading}
-                digestUrl={digestUrl}
-                tags={tags}
-                onTagsChanged={setTags}
-            />
+            <Tabs
+                value={tab}
+                onChange={handleTabChange}
+                classes={tabsClasses}
+                variant="scrollable"
+                scrollButtons="auto"
+            >
+                {!!data.content.text && (
+                    <TabStyle icon={<Subject />} label="Text" classes={tabClasses} />
+                )}
+
+                {hasPreview && (
+                    <TabStyle icon={<PageviewOutlined />} label="Preview" classes={tabClasses} />
+                )}
+
+                {data.content.filetype !== 'folder' && (
+                    <TabStyle icon={<LocalOfferOutlined />} label="Tags" classes={tabClasses} />
+                )}
 
-            <EmailSection doc={data} collection={collection} />
+                {!!data.content && (
+                    <TabStyle icon={<SettingsApplicationsOutlined />} label="Meta" classes={tabClasses} />
+                )}
 
-            <PreviewSection
-                docTitle={data.content.filename}
-                type={data.content["content-type"]}
-                url={docRawUrl}
-            />
+                {data.content.filetype === 'email' && (
+                    <TabStyle icon={<EmailOutlined />} label="Email" classes={tabClasses} />
+                )}
 
-            <HTMLSection html={data.safe_html} />
+                {!!data.safe_html && (
+                    <TabStyle icon={<CodeOutlined />} label="HTML" classes={tabClasses} />
+                )}
 
-            <TextSection
-                title="Text"
-                text={data.content.text}
-                fullPage={fullPage}
-            />
+                {!!data.content.tree && (
+                    <TabStyle icon={<AccountTreeOutlined />} label="Headers &amp; Parts" classes={tabClasses} />
+                )}
 
-            <TextSection
-                title="Headers &amp; Parts"
-                text={data.content.tree}
-                fullPage={fullPage}
-            />
+                {ocrData.map(({tag}) => (
+                    <TabStyle icon={<TextFields />} label={"OCR " + tag} classes={tabClasses} key={tag} />
```

Comment:
Sorry, I haven't specified initially, but the OCR texts need to go as sub-tabs of the Text tab.


Having separate tabs for "Text", "OCR TESSERACT_ENG", "OCR_TESSERACT_FRA" and "OCR CUSTOM 1" will take up all the space available for tabs.


The `text` field and the `ocrtext.$ID` are all different variants of the same text, so we'll be running NLP independently for each of these in the future. That's why I think we would want them tabbed too.


For the implicit text tab, show it under a sub-tab with the title "Extracted from file", so we'll have all texts on the same level.